docker/dotcms/Dockerfile

```diff
@@ -14,71 +15,90 @@ ARG BUILD_ID=HEAD
+ENV DEBIAN_FRONTEND=noninteractive
 
-RUN apk add libsass --no-cache --repository http://dl-cdn.alpinelinux.org/alpine/edge/community/
+# Installing basic packages 
+RUN apt update && \
+    apt upgrade -y 
 
-# This is a nasty hack for libjsass, Alpine 3.9 no longer has a deprecated symlink in /lib
-RUN [ ! -e /lib/ld-linux-x86-64.so.2 ] &&  ln -s $(readlink /lib64/ld-linux-x86-64.so.2) /lib/ld-linux-x86-64.so.2 || :
+# Install prebuilt dockerize
+RUN curl -Ls https://repo.dotcms.com/artifactory/ext-release-local/com/dotcms/dockerize/dockerize-linux-`uname -m` | install /dev/stdin /usr/local/bin/dockerize 
 
-# Download and build dockerize distributed under MIT license (https://github.com/jwilder/dockerize)
-RUN wget --quiet https://github.com/jwilder/dockerize/archive/v0.6.1.tar.gz \
-    && tar -C / -xzf v0.6.1.tar.gz  \
-    && cd /dockerize-0.6.1 \
-    && go get github.com/hpcloud/tail \
-    && go get github.com/jwilder/gojq \
-    && go get golang.org/x/net/context \
-    && go build \
-    && cp dockerize-0.6.1 /usr/local/bin/dockerize \
-    && chmod a+x /usr/local/bin/dockerize
+# Cleanup
+RUN apt purge --allow-remove-essential -y  && \
+    apt autoremove -y && \
+    apt clean && \
+    rm -rf /var/lib/apt/lists/*
 
-RUN cd / && rm -rf  /dockerize-0.6.1
+# Defining default non-root user UID, GID, and name 
+ARG USER_UID="1000000000"
+ARG USER_GID="1000000000"
+ARG USER_NAME="dotcms"
 
-RUN apk del -r --purge git gcc musl-dev go
+# Creating default non-user 
+# the useradd 
+RUN groupadd -g $USER_GID $USER_NAME && \
+	useradd -l -d /srv -g $USER_GID -u $USER_UID $USER_NAME
 
-ENV JAVA_HOME /usr/lib/jvm/default-jvm
-
-COPY --from=build-cms /srv/ /srv/
-COPY ROOT/ /
+# Copy our build
+COPY --from=build-cms --chown=dotcms:dotcms /srv/ /srv/
+COPY --chown=dotcms:dotcms ROOT/ /
 
 # Make scripts runable
-RUN find /srv/ -type f -name "*.sh" -exec chmod a+x {} \;
-
-# Make plugin merging directories writable
-RUN find /srv/templates -type d -exec chmod 770 {} \;
-
-RUN mkdir -p /data/shared/assets && mkdir -p /data/local/dotsecure/license && chmod -R 660 /data && find /data/ -type d -exec chmod 770 {} \;
+ RUN find /srv/ -type f -name "*.sh" -exec chmod a+x {} \; && \
+    # Make plugin merging directories writable
+    find /srv/templates -type d -exec chmod 770 {} \;  && \
+    # Make dotcms user owner
+    mkdir -p /data/shared/assets && \
+    mkdir -p /data/local/dotsecure/license && \
+    chown -R dotcms /data 
+
+# ----------------------------------------------
+# Stage 3:  Flatten everything to 1 layer
+# ----------------------------------------------
+FROM scratch
 
 ENV SERVICE_DELAY_DEFAULT_MIN 1
 ENV SERVICE_DELAY_DEFAULT_STEP 3
 ENV SERVICE_DELAY_DEFAULT_MAX 30
 
-#VOLUME /data/local
+LABEL com.dotcms.contact "support@dotcms.com"
+LABEL com.dotcms.vendor "dotCMS LLC"
+LABEL com.dotcms.description "dotCMS Content Management System"
+
+ARG USER_UID="1000000000"
+ARG USER_GID="1000000000"
+
+COPY --from=container-base / /
```

Comment:
This basically copies the whole linux distribution partition plus the dotcms stuff (from stage #2) that is already there. You are turning an empty layer into a `ubuntu:20.04` + DotCMS, right?

Reply:
Right.  `docker buildx` does not support `--flatten`.   Doing this flattens all layers and turns a 1.2 GB image into a 500mb image.